### PR TITLE
Update to sigstore v0.10.0 to pickup cosign fix

### PR DESCRIPTION
**Pull request by anfredette**
Re-enable cosign signature validation by default.

The sigstore update also included some API changes which required changes in bpfman.

Fixes: #1241

### CODE REVIEW

**Comment by anfredette** [comment]
@dave-tucker can you take a look at this change?  Sigstore changed a couple of APIs including lifetime and async changes.  My updates work, but I wanted to know if you had any concerns.  Thanks, Andre

File: bpfman/src/config.rs [hunk 1]
```diff
@@ -65,7 +65,7 @@ impl Default for SigningConfig {
             // be enabled by default, but it is currently disabled due to the
             // the TUF issue (https://github.com/bpfman/bpfman/issues/1241).
             // This should be set back to true once the issue is resolved.
-            verify_enabled: false,
+            verify_enabled: true,
```
Comment (Billy99) [inline bpfman/src/config.rs:68]: Remove TODO from comment

**Review by dave-tucker — APPROVED** [review]

**Comment by dave-tucker** [comment]
LGTM but added a `hold` label until Billy's comment is addressed.

File: bpfman/src/config.rs (hunk 1 above)
Comment (anfredette) [reply bpfman/src/config.rs:63]: Done

Comment (anfredette) [reply bpfman/src/config.rs:63]: I meant to remove that comment.  Thanks for catching it!

**Review by Billy99 — APPROVED** [review]
/lgtm

**Pull request merged**